modules/channel-web/src/backend/api.ts

```diff
@@ -133,8 +131,17 @@ export default async (bp: typeof sdk, db: Database) => {
         return res.status(400).send(ERR_MSG_TYPE)
       }
 
-      if (timezone && isValidTimezone) {
-        await bp.users.updateAttributes('web', userId, { timezone })
+      if (payload.type === 'visit') {
+        const { timezone } = payload
+        const isValidTimezone = _.isNumber(timezone) && timezone >= -12 && timezone <= 14 && timezone % 0.5 === 0
+
+        const newAttributes = {
+          ...(isValidTimezone && { timezone })
```

Comment:
I'm not a fan of this syntax, feels like an abuse of spread op

Reply:
I need to add another conditional property to the attributes in another PR (the user language), so that syntax would keep it simple